2019-12-07 17:14:11,629 ERROR: Exception on /index [GET] [in C:\anaconda\envs\microblog\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\anaconda\envs\microblog\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "D:\Martin\projects\microblog\app\routes.py", line 28, in index
    page, app.config['POSTS_PE_PAGE'], False)
KeyError: 'POSTS_PE_PAGE'
2019-12-07 17:14:43,694 INFO: Microblog startup [in D:\Martin\projects\microblog\app\__init__.py:47]
2019-12-07 20:27:44,110 INFO: Microblog startup [in D:\Martin\projects\microblog\app\__init__.py:47]
2019-12-07 22:47:11,046 INFO: Microblog startup [in D:\Martin\projects\microblog\app\__init__.py:49]
2019-12-08 22:38:06,602 INFO: Microblog startup [in D:\Martin\projects\microblog\app\__init__.py:49]
2019-12-08 22:42:59,978 ERROR: Exception on /reset_password_request [GET] [in C:\anaconda\envs\microblog\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\Martin\projects\microblog\app\routes.py", line 157, in reset_password_request
    title='Reset Password', form=form)
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\templating.py", line 140, in render_template
    ctx.app,
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\templating.py", line 120, in _render
    rv = template.render(context)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "D:\Martin\projects\microblog\app\templates\reset_password_request.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\environment.py", line 1005, in render
    return concat(self.root_render_func(self.new_context(vars)))
  File "D:\Martin\projects\microblog\app\templates\reset_password_request.html", line 14, in root
    <p>{{ form.submit() }}</p>
  File "D:\Martin\projects\microblog\app\templates\base.html", line 14, in root
    <span class="icon-bar"></span>
  File "C:\anaconda\envs\microblog\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 10, in root
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
  File "C:\anaconda\envs\microblog\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block_doc
    {% block body -%}
  File "C:\anaconda\envs\microblog\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 39, in block_html
  File "C:\anaconda\envs\microblog\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 92, in block_body
  File "D:\Martin\projects\microblog\app\templates\reset_password_request.html", line 23, in block_content
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\runtime.py", line 262, in call
    return __obj(*args, **kwargs)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\runtime.py", line 630, in _fail_with_undefined_error
    raise self._undefined_exception(hint)
jinja2.exceptions.UndefinedError: 'app.forms.ResetPasswordRequestForm object' has no attribute 'hiddent_tag'
2019-12-08 22:44:33,988 ERROR: Exception on /reset_password_request [GET] [in C:\anaconda\envs\microblog\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\Martin\projects\microblog\app\routes.py", line 157, in reset_password_request
    title='Reset Password', form=form)
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\templating.py", line 140, in render_template
    ctx.app,
  File "C:\anaconda\envs\microblog\lib\site-packages\flask\templating.py", line 120, in _render
    rv = template.render(context)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "D:\Martin\projects\microblog\app\templates\reset_password_request.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\environment.py", line 1005, in render
    return concat(self.root_render_func(self.new_context(vars)))
  File "D:\Martin\projects\microblog\app\templates\reset_password_request.html", line 14, in root
    <p>{{ form.submit() }}</p>
  File "D:\Martin\projects\microblog\app\templates\base.html", line 14, in root
    <span class="icon-bar"></span>
  File "C:\anaconda\envs\microblog\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 10, in root
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
  File "C:\anaconda\envs\microblog\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block_doc
    {% block body -%}
  File "C:\anaconda\envs\microblog\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 39, in block_html
  File "C:\anaconda\envs\microblog\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 92, in block_body
  File "D:\Martin\projects\microblog\app\templates\reset_password_request.html", line 23, in block_content
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\runtime.py", line 262, in call
    return __obj(*args, **kwargs)
  File "C:\anaconda\envs\microblog\lib\site-packages\jinja2\runtime.py", line 630, in _fail_with_undefined_error
    raise self._undefined_exception(hint)
jinja2.exceptions.UndefinedError: 'app.forms.ResetPasswordRequestForm object' has no attribute 'hiddent_tag'
2019-12-08 22:44:54,695 INFO: Microblog startup [in D:\Martin\projects\microblog\app\__init__.py:49]
